db/db_local/start_database.go

```diff
@@ -161,6 +156,31 @@ func startPostgresProcess(port int, listen StartListenType, invoker constants.In
 	if err != nil {
 		return err
 	}
+	return nil
+}
+
+// startPostgresProcessAndSetPassword starts the postgres process and writes out the state file
+// after it is convinced that the process is started and is accepting connections
+func startPostgresProcessAndSetPassword(port int, listen StartListenType, invoker constants.Invoker) (e error) {
```

Comment:
this function is way too big
move 
```
postgresCmd := exec.Command(
		getPostgresBinaryExecutablePath(),
		// by this time, we are sure that the port if free to listen to
		"-p", fmt.Sprint(port),
		"-c", fmt.Sprintf("listen_addresses=\"%s\"", listenAddresses),
		// NOTE: If quoted, the application name includes the quotes. Worried about
		// having spaces in the APPNAME, but leaving it unquoted since currently
		// the APPNAME is hardcoded to be steampipe.
		"-c", fmt.Sprintf("application_name=%s", constants.APPNAME),
		"-c", fmt.Sprintf("cluster_name=%s", constants.APPNAME),

		// log directory
		"-c", fmt.Sprintf("log_directory=%s", constants.LogDir()),

		// If ssl is off  it doesnot matter what we pass in the ssl_cert_file and ssl_key_file
		// SSL will only get validated if the ssl is on
		"-c", fmt.Sprintf("ssl=%s", SslStatus()),
		"-c", fmt.Sprintf("ssl_cert_file=%s", filepath.Join(getDataLocation(), constants.ServerCert)),
		"-c", fmt.Sprintf("ssl_key_file=%s", filepath.Join(getDataLocation(), constants.ServerKey)),

		// Data Directory
		"-D", getDataLocation())

	postgresCmd.Env = append(os.Environ(), fmt.Sprintf("STEAMPIPE_INSTALL_DIR=%s", constants.SteampipeDir))
``` 
into a function

and also
```
// create a channel with a big buffer, so that it doesn't choke
	logChannel := make(chan string, 1000)
	stopListenFn, err := setupLogCollector(postgresCmd, logChannel)
	if err == nil {
		defer func() {
			stopListenFn()
		}()
		go traceoutServiceLogs(logChannel)
	} else {
		// this is a convenience and therefore, we shouldn't error out if we
		// are not able to capture the logs.
		// instead, log to TRACE that we couldn't and continue
		log.Println("[TRACE] Warning: Could not attach to service logs")
	}
```